pygbm/splitting.py

```diff
@@ -304,6 +308,59 @@ def split_indices(context, split_info, sample_indices):
     return (sample_indices[:right_child_position],
             sample_indices[right_child_position:])
 
+@njit(parallel=False)
+def split_indices_single_thread(context, split_info, sample_indices):
+    """Split samples into left and right arrays.
+
+    This implementation requires less memory than the parallel version.
```

Comment:
Maybe add a comment about using Hoare's partition scheme, just for reference

Reply:
Ok, I didn't know it had a name, I'm ok with adding it, but did you check that it is effectively the same algo?

Reply:
Yes I'm pretty sure it is https://en.wikipedia.org/wiki/Quicksort#Hoare_partition_scheme